client_auto.go

```diff
@@ -0,0 +1,95 @@
+package hdfs
+
+import (
+	"encoding/xml"
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+)
+
+type Property struct {
+	Name  string `xml:"name"`
+	Value string `xml:"value"`
+}
+
+type Result struct {
+	Property []Property `xml:"property"`
+}
+
+type HadoopConf map[string]string
+
+var ErrUnresolvedNamenode = errors.New("Couldn't find a namenode address in any config.")
+
+// Get Hadoop Properties - try to open a conf file, marshal the results
+// into a Result object and return the Properties of that object.
+func LoadHadoopConfig(path string) (HadoopConf, error) {
+	result := Result{}
+	f, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	xmlErr := xml.Unmarshal(f, &result)
+	if xmlErr != nil {
+		return nil, xmlErr
```

Comment:
you should wrap this error so that the caller gets more information. something like `fmt.Errorf("reading %s: %s", path, xmlErr)` would do it.